conans/client/cmd/export.py

```diff
@@ -144,6 +145,10 @@ def _replace_scm_data_in_conanfile(conanfile_path, scm_data):
     new_text = "scm = " + ",\n          ".join(str(scm_data).split(",")) + "\n"
     content = content.replace(to_replace[0], new_text)
     content = content if not headers else ''.join(headers) + content
+
+    if not os.access(conanfile_path, os.W_OK):
+        os.chmod(conanfile_path, stat.S_IRWXU)  # read, write, execute/search by owner
+
```

Comment:
I would have made a `os.remove(conanfile_path)` before overwritting it, but it fails in Windows with a `PermissionDenied` error 😞

Reply:
I think I would prefer to not alter the conanfile but force the deletion only when present, something similar to the `rmdir` i the `files.py` but for a file. Encapsulated in the files module. It is almost the same but a bit less creepy.

Reply:
AFAIK it is ok to have the `conanfile.py` with those permissions. I've added the helper function, have a look at the last commit.